spyderlib/restart_app.py

```diff
@@ -135,6 +145,29 @@ def main():
         time.sleep(0.2)  # Throttling control
 
     env = os.environ.copy()
+
+    # Reset spyder if needed
+    if reset:
+        # Build reset command
+        command_reset = '"{0}" "{1}" {2}'.format(python, spyder, args_reset)
+
+        # Try to reset
+        try:
+            p = subprocess.Popen(command_reset, shell=shell)
+            p.communicate()
+            pid_reset = p.pid
+        except Exception as error:
+            print(command)
+            traceback.print_exc()
+            time.sleep(15)

```

Comment:
I am not sure I get you... by internal console you mean? at this point we already left the initial spyder instance and we are inside the restart script. I will add an extra flag so that the restart stops and a QMessageBox displays something.